controllers/machine_controller.go

```diff
@@ -614,6 +629,69 @@ func (r *MachineReconciler) shouldAdopt(m *clusterv1.Machine) bool {
 	return metav1.GetControllerOf(m) == nil && !util.HasOwner(m.OwnerReferences, clusterv1.GroupVersion.String(), []string{"Cluster"})
 }
 
+func (r *MachineReconciler) watchClusterNodes(ctx context.Context, cluster *clusterv1.Cluster) error {
+	// If there is no tracker, don't watch remote nodes
+	if r.Tracker == nil {
+		return nil
+	}
+
+	if err := r.Tracker.Watch(ctx, remote.WatchInput{
+		Name:         "machine-watchNodes",
+		Cluster:      util.ObjectKey(cluster),
+		Watcher:      r.controller,
+		Kind:         &corev1.Node{},
+		EventHandler: &handler.EnqueueRequestsFromMapFunc{ToRequests: handler.ToRequestsFunc(r.nodeToMachine)},
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *MachineReconciler) nodeToMachine(o handler.MapObject) []reconcile.Request {
+	node, ok := o.Object.(*corev1.Node)
+	if !ok {
+		r.Log.Error(errors.New("incorrect type"), "expected a Node", "type", fmt.Sprintf("%T", o))
+		return nil
+	}
+
+	machineList := &clusterv1.MachineList{}
+	if err := r.Client.List(
+		context.TODO(),
+		machineList,
+		client.MatchingFields{machineNodeNameIndex: node.Name},
+	); err != nil {
+		r.Log.Error(err, "failed getting machine list")
+		return nil
+	}
+
+	if len(machineList.Items) != 1 {
+		machineNames := make([]string, 0, len(machineList.Items))
+		for _, m := range machineList.Items {
+			machineNames = append(machineNames, m.Name)
+		}
+		r.Log.Error(errors.Errorf("expecting one machine for node %v, got %v", node.Name, machineNames), "Unable to retrieve machine from node", "node", node.GetName())
+		return nil
+	}
+	var requests []reconcile.Request
+	key := util.ObjectKey(&machineList.Items[0])
+	requests = append(requests, reconcile.Request{NamespacedName: key})
+	return requests
```

Comment:
What about 
```suggestion
	key := util.ObjectKey(&machineList.Items[0])
	return []reconcile.Request{ reconcile.Request{NamespacedName: key} }
```
Thus avoiding to append to a slice